ical/parsing/parser.py

```diff
@@ -25,96 +25,133 @@
-    # either be repreated property parameters with the same name or property
-    # s parameters with repeated values. A two level structure is used to grab
-    # both, which is then flattened when consuming the result.
-    param = Group(
-        param_name.set_results_name(PARSE_PARAM_NAME)
-        + "="
-        + param_value.set_results_name(PARSE_PARAM_VALUE, list_all_matches=True)
-        + ZeroOrMore(
-            "," + param_value.set_results_name(PARSE_PARAM_VALUE, list_all_matches=True)
+def parse_line(line: str) -> ParsedProperty:
+    """Parse a single line."""
+    
+    result = ParsedProperty(name="", value="", params=None)
+    line_len = len(line)
+    pos = 0
+
+    # parse NAME
+    while True:
+        if pos >= line_len:
+            raise CalendarParseError(f"Unexpected end of line. Expected ';' or ':'", detailed_error=line)
+        if line[pos] in (';', ':'):
+            name = line[0:pos]
+            if not _RE_NAME.fullmatch(name):
+                raise CalendarParseError(f"Invalid property name '{name}'", detailed_error=line)
+            result.name = name.lower()
+            break
+        pos += 1
+
+    # parse PARAMS if any
+    if line[pos] == ';':
+        result.params = []
+        pos += 1
+        params_start = pos
+        all_params_read = False
+
+        while pos < line_len and not all_params_read:
+            
+            # Read until we hit name/value separator (=)
+            if line[pos] != '=':
+                pos += 1
+                if pos >= line_len:
+                    raise CalendarParseError(f"Unexpected end of line. Expected '='", detailed_error=line)
+                continue
+                
+            # param name reached
+            param_name = line[params_start:pos]
+            if not _RE_NAME.fullmatch(param_name):
+                raise CalendarParseError(f"Invalid parameter name '{param_name}'", detailed_error=line)
+            pos += 1
+            
+            # Now read values. (list separated by comma)
+            param_values = []
+            param_value_start = 0
+            all_values_read = False
+            
+            while not all_values_read:
+                param_value_start = pos
+                quoted = False
+
+                if line[pos] == '"':
+                    # parameter value is quoted
+                    quoted = True
+                    pos += 1
+
+                param_value_read = False
+                while not param_value_read:
+                    if pos >= line_len:
+                        if quoted:
+                            raise CalendarParseError(f"Unexpected end of line. Expected end of qouted string", detailed_error=line)
+                        else:
+                            raise CalendarParseError(f"Unexpected end of line. Expected ',', ';' or ':'", detailed_error=line)
+                    
+                    char = line[pos]
+
+                    if char == '"':
```

Comment:
Can you add an additional test cases in the quoted params file with something like this?

```
NAME;PARAM-NAME="":VALUE
NAME;PARAM-NAME="",Value2,"",VALUE4,"VALUE5":VALUE
```

This catches a bug in existing pyparsing quoted string since it expects more than one character, but the rfc allows it to be empty.